### PR TITLE
native VPs

### PR DESCRIPTION
**Pull request by tzemanovic**
this PR provides an interface for native VPs as described in #300

### CODE REVIEW

File: shared/src/ledger/native_vp.rs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+where
+    DB: storage::DB + for<'iter> storage::DBIter<'iter>,
+    H: StorageHasher,
+{
+    /// Storage prefix iterators.
+    pub iterators: PrefixIterators<'a, DB>,
+    /// VP gas meter.
+    pub gas_meter: VpGasMeter,
+    /// Read-only access to the storage.
+    pub storage: &'a Storage<DB, H>,
```
Comment (yito88) [inline shared/src/ledger/native_vp.rs:67]: I think a native VP stores the data for the functionality to the storage.
Should we provide `write` functions?

**Review by yito88 — COMMENTED** [review]
Thank you! Looks great.
I have a question.

File: shared/src/ledger/native_vp.rs (hunk 1 above)
Comment (tzemanovic) [reply shared/src/ledger/native_vp.rs:67]: Thank you for reviewing! It is my understanding that any storage modifications should be done via transactions (via the write log) and then the VP checks that these are correct (same as with wasm VPs). For the native VPs integrations, we can add functions to write storage in the `vm_env` crate using the tx host env functions. These can then be called from transactions.

Comment (yito88) [reply shared/src/ledger/native_vp.rs:67]: Sorry, I misunderstood that. As we discussed, the VP doesn't update the storage. So, write access isn't needed.

**Review by yito88 — APPROVED** [review]

**Pull request merged**